### PR TITLE
kernel: bump 6.6 to 6.6.47

### PR DESCRIPTION
**Pull request by graysky2**
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.47

All patches automatically rebased.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

### CODE REVIEW

**Comment by graysky2** [comment]
Does anyone know if this new ksym should be set or not and for which targets?

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/diff/arch/arm64/Kconfig?id=v6.6.46&id2=v6.6.45

File: target/linux/bcm27xx/bcm2712/config-6.6 [hunk 1]
```diff
@@ -29,6 +29,7 @@ CONFIG_ARM64_EPAN=y
 CONFIG_ARM64_ERRATUM_1165522=y
 CONFIG_ARM64_ERRATUM_1286807=y
 CONFIG_ARM64_ERRATUM_1463225=y
+CONFIG_ARM64_ERRATUM_3194386=y
```
Comment (hauke) [inline target/linux/bcm27xx/bcm2712/config-6.6:32]: This is not needed for raspberry pi, this workaround is only needed for Cortex-X4.

**Comment by zxlhhyccc** [comment]
image

**Comment by graysky2** [comment]
I had to downgraded to the image I built based on 6.6.45 from 4 days ago due to breakage. Some websites would simply not load while others were slow to load at best and would time out at worst. I do not know the root cause of the problem. I saw nothing in the logs yet this phenomenon was observed on multiple devices. The downgrade restored expected functionality.

I did not add the 6.6.46 kernel to the commit from 4 days ago to verify that some other commit was the cause due to a lack of time to follow-up.

**Comment by DragonBluep** [comment]
> I had to downgraded to the image I built based on 6.6.45 from 4 days ago due to breakage. Some websites would simply not load while others were slow to load at best and would time out at worst. I do not know the root cause of the problem. I saw nothing in the logs yet this phenomenon was observed on multiple devices. The downgrade restored expected functionality. I did not add the 6.6.46 kernel to the commit from 4 days ago to verify that some other commit was the cause due to a lack of time to follow-up.
> […](#)
> Message ID: ***@***.***>

Can you try reverting this patch?
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit?h=v6.6.46&id=6772c4868a8e7ad5305957cdb834ce881793acb7

**Comment by curtdept** [comment]
> > I had to downgraded to the image I built based on 6.6.45 from 4 days ago due to breakage. Some websites would simply not load while others were slow to load at best and would time out at worst. I do not know the root cause of the problem. I saw nothing in the logs yet this phenomenon was observed on multiple devices. The downgrade restored expected functionality. I did not add the 6.6.46 kernel to the commit from 4 days ago to verify that some other commit was the cause due to a lack of time to follow-up.
> > […](#)
> > Message ID: _**@**_.***>
> 
> Can you try reverting this patch? https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit?h=v6.6.46&id=6772c4868a8e7ad5305957cdb834ce881793acb7
> 
> Edit: I didn't built the latest kernel myself.

There is some other kernel patch that is interacting with this one causing BAD packetloss. I cut my patches out (im on x86) and  went back to vanilla and its perfect.

**Comment by graysky2** [comment]
Did you build this commit against a last-known-good commit such that this is the only change?

**Comment by drujd** [comment]
I have cherrypicked this on top of current master (which seems to be only 1 commit ahead of this pr's base). After DNS mostly not working over ethernet, I went back to the master and everything is fine. So yeah, 100% caused by this kernel upgrade, probably bad interaction with one of the openwrt patches (assuming @curtdept is right).

File: target/linux/bcm27xx/bcm2712/config-6.6 (hunk 1 above)
Comment (hauke) [reply target/linux/bcm27xx/bcm2712/config-6.6]: sorry, you were right the Cortex-A76 in the Raspberry Pi 5 is also affected.

Comment (hauke) [reply target/linux/bcm27xx/bcm2712/config-6.6]: on the armsr target we should also activate this option.

**Comment by graysky2** [comment]
> Can you try reverting this patch?
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit?h=v6.6.46&id=6772c4868a8e7ad5305957cdb834ce881793acb7

@DragonBluep - Reverted and building now.  Will report back.

File: target/linux/bcm27xx/bcm2712/config-6.6 (hunk 1 above)
Comment (graysky2) [reply target/linux/bcm27xx/bcm2712/config-6.6:32]: Done, see latest commit

**Comment by graysky2** [comment]
@curtdept and others - If you're correct about one or more of OpenWrt's patches causing the issue, I think the right course of action would be to not revert the upstream commit to 6.6.46 but to figure out which of our patches is causing the problem.  Am I correct or is a better option to include yet another hack-6.6 patch to revert the upstream commit called out by @DragonBluep?

I am building/testing this HEAD + PR + a revert patch now.

**Comment by graysky2** [comment]
I do not believe that reverting that upstream commit fixed the issue for me.  Upon booting into the image, I am noticing the same sluggish behavior of some websites simply not loading.  Can someone else confirm?  [Here](https://gist.github.com/graysky2/c2a53e541daa5dfb6d4630ba8cd6555c) is a patch to apply to your build system to introduce the revert.

**Comment by graysky2** [comment]
>I cut my patches out (im on x86) and went back to vanilla and its perfect.

@curtdept - Which patches did you remove?  Can you call out the dir tree/trees?

**Comment by curtdept** [comment]
> > I cut my patches out (im on x86) and went back to vanilla and its perfect.
> 
> @curtdept - Which patches did you remove? Can you call out the dirtree/trees?

I took a lot of heat and the tailscale guys produced good rersults when I pushed back on the kernel maintainers about the bad gso patch which is what led me to this...

I wish this was going to be more helpful, also why I havent been more specific but here goes:

- all patches removed in target/linux/generic/pending-6.6 except for 103-kbuild-export-SUBARCH.patch and 100-compiler.h-only-include-asm-rwonce.h-for-kernel-code.patch
- all patches removed in target/linux/generic/hack-6.6 except 205-kconfig-abort-configuration-on-unset-symbol.patch
- all patches removed in target/linux/generic/backport-6.6

Its amazing just how many of these don't apply at all, I can tell you my build is a boatload more responsive as well.

Sorry I know that's basically a needle in a haystack. But I def have a good build on 6.6.45 with the patch and 6.6.46.

**Comment by lorand-horvath** [comment]
@graysky2 @curtdept I guess you've already noticed or know about @nbd168 's recent commits: https://github.com/openwrt/openwrt/commit/79fce424046fed521026df7c7f838441fb46add8 and https://github.com/openwrt/openwrt/commit/17f07040e7808a0c9cba2afc0e0b22e5589da133

**Comment by graysky2** [comment]
> I guess you've already noticed or know about @nbd168 's recent commits: https://github.com/openwrt/openwrt/commit/79fce424046fed521026df7c7f838441fb46add8 and https://github.com/openwrt/openwrt/commit/17f07040e7808a0c9cba2afc0e0b22e5589da133

I rebased/refreshed the PR but upon booting into it, I am still experiencing the same behavior I reported initially.  Can anyone else confirm?

**Comment by vincejv** [comment]
I only cherry-pick kernel-related patches/commits, I don't have issues with or without https://github.com/openwrt/openwrt/commit/79fce424046fed521026df7c7f838441fb46add8 and https://github.com/openwrt/openwrt/commit/17f07040e7808a0c9cba2afc0e0b22e5589da133

No noticeable changes in browsing experience, even with QUIC/HTTP3 websites, this is my graph with node exporter dashboard
image

i even host an http/3 site myself + matrix synapse server (which processes a lot of incoming and outgoing http2 or 3 requests) through docker, have no issues so far

using pppoe on Intel 82579LM NIC for WAN and r8169 driver for LAN, i did notice an **increase in http3 throughput from 60 to 110mbps on librespeed server** (the total server bandwidth is 300mbps) hosted through nginx after these commits https://github.com/openwrt/openwrt/commit/79fce424046fed521026df7c7f838441fb46add8 and https://github.com/openwrt/openwrt/commit/17f07040e7808a0c9cba2afc0e0b22e5589da133

**Comment by graysky2** [comment]
@drujd - can you test this PR now against HEAD and report back?  Still finding dropped DNS requests?

**Comment by drujd** [comment]
> @drujd - can you test this PR now against HEAD and report back? Still finding dropped DNS requests?

@graysky2 yes, still the same problem. Some incoming DNS packets are just dropped.

With the PR, queries for www.seznam.czzzz are dropped, query for signaler-pa.clients6.google.com works perfectly well. The queries with the bad checksum are generated from linux ping command, the working query is from Firefox:
```
tcpdump: listening on br-lan, link-type EN10MB (Ethernet), snapshot length 262144 bytes
10:18:51.464431 IP (tos 0x0, ttl 64, id 56301, offset 0, flags [DF], proto UDP (17), length 96)
    192.168.1.3.56077 > omnia.lan.53: [bad udp cksum 0xe390 -> 0x91ee!] 41494+ A? www.seznam.czzzz. (68)
10:18:56.471661 IP (tos 0x0, ttl 64, id 56303, offset 0, flags [DF], proto UDP (17), length 96)
    192.168.1.3.56077 > omnia.lan.53: [bad udp cksum 0xe390 -> 0x91ee!] 41494+ A? www.seznam.czzzz. (68)
10:18:57.066219 IP (tos 0x0, ttl 64, id 14179, offset 0, flags [DF], proto UDP (17), length 77)
    192.168.1.3.47784 > omnia.lan.53: [udp sum ok] 51596+ HTTPS? signaler-pa.clients6.google.com. (49)
10:18:57.173445 IP (tos 0x0, ttl 64, id 19770, offset 0, flags [none], proto UDP (17), length 127)
    omnia.lan.53 > 192.168.1.3.47784: [bad udp cksum 0x83d1 -> 0x02c2!] 51596 q: HTTPS? signaler-pa.clients6.google.com. 0/1/0 ns: google.com. SOA ns1.google.com. dns-admin.google.com. 663667291 900 900 1800 60 (99)
```

Without the PR, everything works, regardless of checksums. Again, the seznam.czzzz are from running a ping command, the query with correct checksum is from Spotify:
```
tcpdump: listening on br-lan, link-type EN10MB (Ethernet), snapshot length 262144 bytes
10:21:27.196476 IP (tos 0x0, ttl 64, id 44039, offset 0, flags [DF], proto UDP (17), length 96)
    192.168.1.3.49408 > omnia.lan.53: [bad udp cksum 0xcbed -> 0x596e!] 54214+ A? www.seznam.czzzz. (68)
10:21:27.228922 IP (tos 0x0, ttl 64, id 2345, offset 0, flags [none], proto UDP (17), length 137)
    omnia.lan.53 > 192.168.1.3.49408: [bad udp cksum 0x83db -> 0x6cc8!] 22213 NXDomain$ q: AAAA? www.seznam.czzzz. 0/1/0 ns: . SOA a.root-servers.net. nstld.verisign-grs.com. 2024081601 1800 900 604800 86400 (109)
10:21:27.257468 IP (tos 0x0, ttl 64, id 2346, offset 0, flags [none], proto UDP (17), length 137)
    omnia.lan.53 > 192.168.1.3.49408: [bad udp cksum 0x83db -> 0xefe1!] 54214 NXDomain$ q: A? www.seznam.czzzz. 0/1/0 ns: . SOA a.root-servers.net. nstld.verisign-grs.com. 2024081601 1800 900 604800 86400 (109)
10:21:27.257857 IP (tos 0x0, ttl 64, id 2953, offset 0, flags [DF], proto UDP (17), length 104)
    192.168.1.3.56132 > omnia.lan.53: [bad udp cksum 0x768a -> 0x68eb!] 13433+ A? www.seznam.czzzz.lan. (76)
10:21:27.258629 IP (tos 0x0, ttl 64, id 2347, offset 0, flags [none], proto UDP (17), length 66)
    omnia.lan.53 > 192.168.1.3.56132: [bad udp cksum 0x8394 -> 0xf626!] 13433 NXDomain q: A? www.seznam.czzzz.lan. 0/0/0 (38)
10:21:27.258728 IP (tos 0x0, ttl 64, id 2348, offset 0, flags [none], proto UDP (17), length 66)
    omnia.lan.53 > 192.168.1.3.56132: [bad udp cksum 0x8394 -> 0x130d!] 6008 NXDomain q: AAAA? www.seznam.czzzz.lan. 0/0/0 (38)
10:22:29.153959 IP (tos 0x0, ttl 64, id 12393, offset 0, flags [DF], proto UDP (17), length 71)
    192.168.1.3.58839 > omnia.lan.53: [udp sum ok] 1361+ A? gew4-spclient.spotify.com. (43)
10:22:29.154123 IP (tos 0x0, ttl 64, id 4751, offset 0, flags [none], proto UDP (17), length 125)
    omnia.lan.53 > 192.168.1.3.58839: [bad udp cksum 0x83cf -> 0x77fa!] 1361 q: A? gew4-spclient.spotify.com. 2/0/0 gew4-spclient.spotify.com. CNAME edge-web-gew4.dual-gslb.spotify.com., edge-web-gew4.dual-gslb.spotify.com. A 35.186.224.28 (97)
```

This is weird. With the PR, it seems that incoming packets with bad checksums are dropped and not responded to, while responding to packets whose checksum is OK. Without the PR, the router correctly (?) responds to all requests. Obviously, with hardware checksumming, having bad checksums is to be expected (but I typically see those set to zeroes, not non-zero wrong values) especially on the outbound packets (responses), but why do some incoming packets have correct checksums and some incorrect (both with and without the PR)? Note that it doesn't matter whether I use the DNS server running on the router or some external one (1.1.1.1...).

**Comment by vincejv** [comment]
is www.seznam.czzzz even a registered domain? i've checked various whois sources, it's not registered..

**Comment by drujd** [comment]
No. But that's completely besides the point. It doesn't matter at all whether it's an existing domain or not.

**Comment by drujd** [comment]
So the difference between the requests from browser apps that work and from system/console apps that don't is that the console apps always send 2 requests: One for A and another for AAAA. I can clearly see that in Wireshark (on the side of the computer connected to the router). HOWEVER, tcpdump on the router sees only a single packet of larger length than just a single request (and what I see in Wireshark). As long as I am running without the PR, I can see two separate responses for the A and the AAAA query, so clearly both are received and responded to correctly.

Is there some optimisation that joins two frames with identical header happening that breaks this?

**Comment by vincejv** [comment]
okay i can see now the intermittent "slow" internet, but it's only reproducible in my CDC-ECM tethering interfaces, can't repro it over on PPPoE...

**Comment by curtdept** [comment]
> okay i can see now the intermittent "slow" internet, but it's only reproducible in my CDC-ECM tethering interfaces, can't repro it over on PPPoE...

I'm on pppoe (centurylink) was happening, easiest way for me to see was the loss indicator on speed.cloudflare.com. would show as 90%+ loss

**Comment by drujd** [comment]
All my DNS issues are resolved by removing `target/linux/generic/hack-6.6/600-net-enable-fraglist-GRO-by-default.patch`. Without that patch, I see all incoming packets separately with correct lengths and correct checksums. So there is some breakage regarding 'fraglist GRO'.

**Comment by zxlhhyccc** [comment]
> All my DNS issues are resolved by removing `target/linux/generic/hack-6.6/600-net-enable-fraglist-GRO-by-default.patch`. Without that patch, I see all incoming packets separately with correct lengths and correct checksums. So there is some breakage regarding 'fraglist GRO'.

Yes.

**Comment by graysky2** [comment]
> Yes. delete target/linux/generic/hack-6.6/600-net-enable-fraglist-GRO-by-default.patch problem solved.

@nbd168 - Felix, this is your patch.  Can you comment?

**Comment by zxlhhyccc** [comment]
@graysky2 Compared to the 6.1.105 kernel (which has this patch and is currently running fine), this patch actually needs to be restored：
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit?h=v6.6.46&id=6772c4868a8e7ad5305957cdb834ce881793acb7

**Comment by graysky2** [comment]
@zxlhhyccc - I don't understand what you mean when you say "restored"... it is currently applied to 6.6.46.  Are you saying you believe it needs to be reverted?

**Comment by graysky2** [comment]
> easiest way for me to see was the loss indicator on speed.cloudflare.com. would show as 90%+ loss.

@curtdept - This is an excellent way to see the problem manifest for me as well.

**Comment by namiltd** [comment]
Version 6.6.47 has just been released.

**Comment by namiltd** [comment]
I have doubts about removing 600-net-disable-GRO-by-default.patch
Will it not affect performance?

**Comment by graysky2** [comment]
Umm.. have you read the reports of terrible performance/dropped packets with the patch + news kernel???

**Comment by namiltd** [comment]
This applies to 600-net-enable-fraglist-GRO-by-default.patch
I am writing about 600-net-disable-GRO-by-default.patch. Patch failed but it can be fixed instead of removed.

**Comment by drujd** [comment]
That one is broadcom only. Removing it will keep GRO itself enabled and fraglist disabled (as of the current state of the PR). Indeed the patch should probably be kept and updated to just disable GRO as fraglist is now disabled everywhere with this PR. Ideally, someone with the bcm router should conduct a performance test, disabling fraglist might change things regarding checksumming performance etc.

**Comment by drujd** [comment]
I will try to test the GRO fragliat later today on my PC running vanilla kernel. This might easily be a vanilla kernel regression regarding GRO fraglist, it's just that it's not enabled by default, so nobody noticed/cares.

**Comment by zxlhhyccc** [comment]
@graysky2 `bcm53xx/patches-6.6/600-net-disable-GRO-by-default.patch` can manually rebased？

**Comment by drujd** [comment]
@graysky2
```
--- a/include/linux/netdev_features.h
+++ b/include/linux/netdev_features.h
@@ -243,10 +243,10 @@ static inline int find_next_netdev_featu
 #define NETIF_F_UPPER_DISABLES	NETIF_F_LRO
 
 /* changeable features with no special hardware requirements */
-#define NETIF_F_SOFT_FEATURES	(NETIF_F_GSO | NETIF_F_GRO)
+#define NETIF_F_SOFT_FEATURES	(NETIF_F_GSO)
 
 /* Changeable features with no special hardware requirements that defaults to off. */
-#define NETIF_F_SOFT_FEATURES_OFF	(NETIF_F_GRO_FRAGLIST | NETIF_F_GRO_UDP_FWD)
+#define NETIF_F_SOFT_FEATURES_OFF	(NETIF_F_GRO | NETIF_F_GRO_FRAGLIST | NETIF_F_GRO_UDP_FWD)
 
 #define NETIF_F_VLAN_FEATURES	(NETIF_F_HW_VLAN_CTAG_FILTER | \
 				 NETIF_F_HW_VLAN_CTAG_RX | \
```

**Comment by graysky2** [comment]
Thanks @zxlhhyccc

File: target/linux/bcm53xx/patches-6.6/600-net-disable-GRO-by-default.patch [hunk 2]
```diff
@@ -21,15 +21,15 @@ Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
...
 +++ b/include/linux/netdev_features.h
-@@ -242,10 +242,10 @@ static inline int find_next_netdev_featu
+@@ -243,10 +243,10 @@ static inline int find_next_netdev_featu
  #define NETIF_F_UPPER_DISABLES	NETIF_F_LRO
  
  /* changeable features with no special hardware requirements */
--#define NETIF_F_SOFT_FEATURES	(NETIF_F_GSO | NETIF_F_GRO | NETIF_F_GRO_FRAGLIST)
-+#define NETIF_F_SOFT_FEATURES	(NETIF_F_GSO)
+-#define NETIF_F_SOFT_FEATURES	(NETIF_F_GSO | NETIF_F_GRO)
++#define NETIF_F_SOFT_FEATURES	(NETIF_F_GSO )
```
Comment (zxlhhyccc) [inline target/linux/bcm53xx/patches-6.6/600-net-disable-GRO-by-default.patch:29]: There should be no Spaces.

**Review by zxlhhyccc — COMMENTED** [review]
There should be no Spaces.

**Comment by drujd** [comment]
These kinds of kernel hacks that just change default values... Honestly I don't like them at all, useless patch bloat that makes updating kernel more difficult. This is what https://man7.org/linux/man-pages/man2/ioctl.2.html is for.

File: target/linux/bcm53xx/patches-6.6/600-net-disable-GRO-by-default.patch (hunk 2 above)
Comment (graysky2) [reply target/linux/bcm53xx/patches-6.6/600-net-disable-GRO-by-default.patch]: Typo, fixed, thanks.

**Comment by nbd168** [comment]
Please test if this change fixes the GRO regression in the 6.6.46 update: https://nbd.name/p/b62f69e7

**Comment by zxlhhyccc** [comment]
@nbd168 Do not delete `target/linux/generic/hack-6.6/600-net-enable-fraglist-GRO-by-default.patch` test?

**Comment by drujd** [comment]
> Do not delete `target/linux/generic/hack-6.6/600-net-enable-fraglist-GRO-by-default.patch` test?

If you have ethtool installed, you can always enable GRO_FRAGLIST by `ethtool -K IFNAME rx-gro-list on` regardless of the patch.

**Comment by nbd168** [comment]
Tests should be with 600-net-enable-fraglist-GRO-by-default.patch in place to ensure that GRO gets turned on everywhere.

**Comment by zxlhhyccc** [comment]
@graysky2 May add https://nbd.name/p/b62f69e7  and restore `target/linux/generic/hack-6.6/600-net-enable-fraglist-GRO-by-default.patch`  test？

**Comment by drujd** [comment]
@nbd168 Yep, that fixed it for me!

**Comment by graysky2** [comment]
@nbd168 
> Please test if this change fixes the GRO regression in the 6.6.46 update: https://nbd.name/p/b62f69e7

I restored the deleted patch (`target/linux/generic/hack-6.6/600-net-enable-fraglist-GRO-by-default.patch`) and applied your test commit.  I do not see the bug now.  Will you push that commit to master?  If so, I will restore the patch in this PR.

**Comment by graysky2** [comment]
@nbd168 
> Please test if this change fixes the GRO regression in the 6.6.46 update: https://nbd.name/p/b62f69e7

I restored the deleted patch (`target/linux/generic/hack-6.6/600-net-enable-fraglist-GRO-by-default.patch`) and applied your test commit.  I do not see the bug now.  I added your patch to this PR and restored the aforementioned patches as well.

**Comment by lorand-horvath** [comment]
@graysky2 Will you bump this PR to kernel 6.6.47, since it's already out?

**Comment by zxlhhyccc** [comment]
@graysky2 `bcm53xx/patches-6.6/600-net-disable-GRO-by-default.patch` It also needs to be restored.

**Comment by graysky2** [comment]
@zxlhhyccc - yes, I am doing so now.
@lorand-horvath - yes, once I get 6.6.46 correct, I will add 6.6.47

**Comment by zxlhhyccc** [comment]
@graysky2 
````
target/linux/bcm53xx/patches-6.6/600-net-disable-GRO-by-default.patch
Kernel patches for bcm53xx/generic require refresh. (run 'make target/linux/refresh' and force push this pr)
You can also check the provided artifacts with the refreshed patch from this CI run.
Error: Process completed with exit code 1.
`````

**Comment by graysky2** [comment]
I see it, refreshing it

**Comment by nbd168** [comment]
@graysky2 please move my patch to pending-6.6 instead of hack-6.6 (I already sent it to netdev).
Also, please merge it into the commit that updates the kernel in order to prevent bisect issues.

**Comment by graysky2** [comment]
Isn't pending reserved for patches already accepted into upstream queue?  To my knowledge this one has not been submitted upstream.

**Comment by zxlhhyccc** [comment]
@graysky2 
```
target/linux/generic/pending-6.6/680-net-add-TCP-fraglist-GRO-support.patch
Kernel patches for apm821xx/nand require refresh. (run 'make target/linux/refresh' and force push this pr)
You can also check the provided artifacts with the refreshed patch from this CI run.
Error: Process completed with exit code 1.
````

**Comment by drujd** [comment]
6.6.47 seems to run well on my Turris Omnia.

**Comment by Gingernut1978** [comment]
6.6.47 running fine on my X86_64 box.

**Comment by mgz0227** [comment]
On my X86_64, I used the router as an AP to transmit WiFi, but the sound was intermittent when using iPad for airplay

**Comment by mgz0227** [comment]
On my X86_64, I used the router as an AP to transmit WiFi, but the sound was intermittent when using iPad for airplay

**Pull request merged**